src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1468,11 +1470,58 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
         /// <param name="cancellationToken">To cancel the emit process.</param>
         public EmitResult Emit(
             Stream peStream,
+            Stream pdbStream,
+            Stream xmlDocumentationStream,
+            Stream win32Resources,
+            IEnumerable<ResourceDescription> manifestResources,
+            EmitOptions options,
+            CancellationToken cancellationToken)
+        {
```

Comment:
I believe the pattern for adding optional parameters to already shipped API is to create a backward compat overload with the exact signature as was the shipped method (this one) and add a new overload with the optional parameters. @gafter is that correct?